### DISCUSSION TITLE
Error handling for htmx.ajax()

### DISCUSSION THREAD

**Discussion by gapop** [bigskysoftware/htmx#926] [Q&A] [answered]
When using `htmx.ajax()`, is there a way to detect when the request has failed, for example with a 500 response code from the server? Chaining a `.catch()` method to the promise works for a network timeout, but not for an error response. Listening to the `htmx:reponseError` event (on `document.body`), I can see the error, but it feels weird to use a global event handler for local logic.

**Comment by gapop** [comment, marked as answer]
In case someone else needs this: pass the `handler` callback in the `context` parameter of the `.ajax()` method. This callback should take a parameter, let’s call it `response`, and look for `response['htmx-internal-data'].xhr.status`.

**Reply by alexandredesane** [reply]
hero
